### PR TITLE
New/fish support

### PR DESCRIPTION
**Pull request by mbercx**

### CODE REVIEW

**Review by GeigerJ2 — APPROVED** [review]
Let's unleash it on the world :D

File: aiida_project/project/__init__.py
[the remark this answers is not in the document]
Comment (mbercx) [reply aiida_project/project/__init__.py:1]: Hmm, yeah, I guess I can also move everything into `base.py`. Let me still do that.

**Comment by mbercx** [comment]
We should probably write some tests at some point 😅

File: aiida_project/data/shell_fields.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        set value (echo $line | cut -d= -f2 | sed "s/'//g")
+        set -gx $key $value
+    end
+    function cda
+        source "$aiida_venv_dir/$argv[1]/bin/activate.fish"
+        cd "$aiida_project_dir/$argv[1]"
+    end
+    funcsave -q cda
+  activate: |
+    set -Ux AIIDA_PATH {env_file_path}
```
Comment (mbercx) [inline aiida_project/data/shell_fields.yaml:36]: @GeigerJ2 I think we have to remove the `U` here, no? Else the variable definition seems to be added to the `~/.config/fish/fish_variables` file and will be set for all shell instances.

```suggestion
    set -x AIIDA_PATH {env_file_path}
```

Comment (mbercx) [reply aiida_project/data/shell_fields.yaml:36]: Ah, seems we should be using the `-g` "global" variable setting, not the "universal" one:

https://fishshell.com/docs/current/cmds/set.html

Comment (GeigerJ2) [reply aiida_project/data/shell_fields.yaml:36]: Yeah, you're right. Should be `-g`, not `-U` indeed!

**Review by GeigerJ2 — APPROVED** [review]
Thanks for the refactor! Should we merge it, then? I'm inclined to take a look at the other issues, especially the `conda` implementation :3

**Pull request merged**